### DISCUSSION TITLE
Can Rewrites be applied to dev-only or prod-only?

### DISCUSSION THREAD

**Discussion by opqpop** [vercel/next.js#42717] [Help]
### Describe the feature you'd like to request

Use-case:
- REST API backend is on 'http://localhost:8000/...' while prod is on 'https://153.252.353.22:8000/...' (example host)
- In dev env, we want to hit localhost:8000, as well as have ability to quickly switch to hit 153.252.353.22:8000 if want to test vs prod backend

What is the recommended way to do this? 

Option 1: 

Should we rely on `.env.local` and have every dev add this? This is strange since it introduces complexity in prod to satisfy requirements in dev. 
```
NEXT_PUBLIC_BACKEND_HOST='http://localhost:8000'
```

Option 2:
Should we have every dev add in rewrites into NextJS? Ideally we can make the rewrites apply to dev-only.

Other options?

Current workaround:
- every dev adds a rewrite like this, while prod doesn't have it. But this risks accidentally being checked in and messing up prod if anyone ever tries to check-in next.config.js changes

```
  async rewrites() {
    return [
      {
        source: '/api/:path*',
        destination: 'http://localhost:8000/api/:path*', // Proxy to Backend
      },
    ]
  },
```

### Describe the solution you'd like

dev-only rewrites, or alternative solution to this common problem

### Describe alternatives you've considered

see feature request

**Comment by ijjk** [comment]
Hi, you can conditionally return rewrites specific to dev or production by checking `process.env.NODE_ENV === 'production'` e.g. 

```js
rewrites() {
  return [
    ...(process.env.NODE_ENV !== 'production' ? [ /* devRewrites here */ ] : [])
  ]
}
```
